src/assets/styles/styles.11ty.js

```diff
@@ -0,0 +1,188 @@
+
+Licensed under the New BSD license. You may not use this file except in compliance with this License.
+
+You may obtain a copy of the New BSD License at
+https://github.com/fluid-project/fluidic-11ty/raw/master/LICENSE.md.
+*/
+"use strict";
+
+const fs = require("fs");
+const fg = require("fast-glob");
+const path = require("path");
+const sass = require("sass");
+const stylus = require("stylus");
+const autoprefixer = require("autoprefixer");
+const postcss = require("postcss");
+const CleanCSS = require("clean-css");
+const cssesc = require("cssesc");
+
+const isProd = process.env.ELEVENTY_ENV === "production";
+
+/**
+ * @see https://github.com/maxboeck/eleventastic/blob/master/src/assets/styles/styles.11ty.js
+ */
+
+module.exports = class {
+    // Configure the transform.
+    async data() {
+        const entryPaths = await fg("./src/assets/styles/style.{css,scss,styl}");
+        const entryPath = entryPaths[0];
+        return {
+            permalink: '/assets/styles/style.css',
+            eleventyExcludeFromCollections: true,
+            entryPath
+        };
+    }
+
+    // Compile SCSS to CSS.
+    async compileScss(options) {
+        return new Promise((resolve, reject) => {
+            const config = {
+                file: options.entryPath,
+                sourceMap: !isProd ? 'out.map' : false,
+                sourceMapEmbed: !isProd || false,
+                outputStyle: isProd ? 'compressed' : 'expanded'
+            };
+
+            sass.render(config, (err, result) => {
+                if (err) {
+                    reject(err);
+                } else {
+                    const processedCss = this.processCss({str: result.css, entryPath: options.entryPath});
+                    resolve(processedCss);
+                }
+            });
+        });
+    }
+
+    // Compile Stylus to CSS.
+    async compileStylus(options) {
+        return new Promise((resolve, reject) => {
+            const sourceMap = isProd ? false : { inline: true };
+
+            stylus(options.str)
+                .set('filename', options.entryPath)
+                .set('sourcemap', sourceMap)
+                .set('compress', isProd)
+                .render((err, css) => {
+                    if (err) {
+                        reject(err);
+                    } else {
+                      const processedCss = this.processCss({str: css, entryPath: options.entryPath});
+                      resolve(processedCss);
+                    }
+                });
+        });
+    }
+
+    // Process CSS with PostCSS.
+    async processCss(options) {
+      const from = options.entryPath || 'undefined';
+      const css = postcss([autoprefixer]).process(options.str, {from}).css;
+      return css;
+    }
+
+    // Compile and process CSS (source-agnostic).
+    async compile(options) {
+      let css;
+      switch (options.fileType) {
+        case 'css':
+          css = await this.processCss({str: options.str});
```

Comment:
Is it possible to just pass in the options object here, as it also has a top level `str` property, or do other properties conflict?

Reply:
I figured out a way to simplify this and use `options` directly.